storage/watchable_store_test.go

```diff
@@ -185,6 +186,39 @@ func TestSyncWatchings(t *testing.T) {
 	}
 }
 
+// TestRangeAllUnsynced ...
+func TestRangeAllUnsynced(t *testing.T) {
+	s := &watchableStore{
+		store:    newStore(tmpPath),
+		unsynced: make(map[*watching]struct{}),
+		synced:   make(map[string]map[*watching]struct{}),
+	}
+
+	defer func() {
+		s.store.Close()
+		os.Remove(tmpPath)
+	}()
+
+	for k := 0; k < 2; k++ {
+		testKey := []byte(fmt.Sprintf("foo%d", k))
+		testValue := []byte("bar")
+		s.Put(testKey, testValue)
+
+		w := s.NewWatcher()
+
+		watcherN := 10
+		for i := 0; i < watcherN; i++ {
+			// use 1 to keep watchers in unsynced
+			w.Watch(testKey, true, 1)
+		}
+	}
+	a, b, c, d := s.RangeAllUnsynced()
+	fmt.Println(a)
+	fmt.Println(b)
+	fmt.Println(c)
+	fmt.Println(d)
```

Comment:
this is not testing anything... Just trying to understand what's going on here. Will change it to test.